UI/adsList.js

```diff
@@ -0,0 +1,649 @@
+class AdsList{
+    _adList = [];
+    constructor(adList){
+        this._adList = adList.concat();
+    }
+
+    getPage(skip = 0, top = 10, filterConfig = undefined){
+        if (typeof skip !== 'number' || typeof top !== 'number') {
+            console.log('Error with inputting types!');
+            return;
+        }
+    
+        let returningAds = this._adList;
+        if (filterConfig){
+        
+            for (let parameter in filterConfig){
+                if (parameter === 'hashTags'){
+                    filterConfig.hashTags.forEach(tag => {
+                        returningAds = returningAds.filter(ad => ad.hashTags.includes(tag));
+                    });
+                }
+                if (parameter === 'dateFrom'){
+                    returningAds = returningAds.filter(ad => ad.createdAt >= filterConfig.dateFrom);
+                }
+                else if (parameter === 'dateTo'){
+                    returningAds = returningAds.filter(ad => ad.createdAt < filterConfig.dateTo);
+                }
+                else if (parameter === 'vendor'){
+                    returningAds = returningAds.filter(ad => ad.vendor === filterConfig.vendor);
+                }
+            }
+        }
+            
+        returningAds.sort(this._comparator);
+        return returningAds.slice(skip, skip + top);
+    
+    }
+
+    _comparator(first, second){
+        return second - first;
```

Comment:
I guess, `second.createdAt - first.createdAt`, otherwise you compare links to objects